### PR TITLE
[MRG] MNT Fix meta-test distinguish skipped from other records

### PR DESCRIPTION
**Pull request by jeremiedbb**
Actually check that the test is skipped.
Before that, a false positive could be due to a warning raised in a test for instance.

Also made the meta test way more verbose

### CODE REVIEW

File: continuous_integration/check_no_test_skipped.py [hunk 1]
```diff
@@ -16,18 +16,26 @@
...
+            print("  -", test_name)
+
+            for child in test:
+                print("    -", child.tag)
+                if child.tag == "skipped":
+                    aggregated_results[test_name] |= False
+                    break
             else:
-                aggregated_results[test_name] |= not bool(len(test))
+                aggregated_results[test_name] |= True
```
Comment (ogrisel) [inline continuous_integration/check_no_test_skipped.py]:
```suggestion
                aggregated_results[test_name] = True
```

would be even simpler right?

Comment (jeremiedbb) [reply continuous_integration/check_no_test_skipped.py]: yes :)

**Comment by jeremiedbb** [comment]
> Printing all the tests is going to be really verbose. Wouldn't it be enough to only print the (sorted) skipped test for each jobs?

I guess we can do that now that I've understood what was going on

File: continuous_integration/check_no_test_skipped.py [hunk 2]
```diff
@@ -11,26 +11,34 @@
...
+                    break
             else:
-                aggregated_results[test_name] |= not bool(len(test))
+                always_skipped[test_name] = False
 
+print("\n------------------------------------------------------------------\n")
 fail = False
-for test, result in aggregated_results.items():
-    if not result:
+for test, skipped in always_skipped.items():
```
Comment (ogrisel) [inline continuous_integration/check_no_test_skipped.py]: Not sure if necessary, but never hurts:

```suggestion
for test, skipped in sorted(always_skipped.items()):
```

Comment (jeremiedbb) [reply continuous_integration/check_no_test_skipped.py]: hum why do you want to reorder the tests ? I find it easier is they keep the order they are run in the test suite.

Comment (ogrisel) [reply continuous_integration/check_no_test_skipped.py]: Alright.

File: continuous_integration/check_no_test_skipped.py [hunk 3]
```diff
@@ -11,26 +11,34 @@
...
+                always_skipped[test_name] = True
+
+            for child in test:
+                if child.tag == "skipped":
+                    print("    -", test_name)
+                    always_skipped[test_name] &= True
+                    break
             else:
-                aggregated_results[test_name] |= not bool(len(test))
+                always_skipped[test_name] = False
```
Comment (ogrisel) [inline continuous_integration/check_no_test_skipped.py]: What about?

```python
        for test in root.iter("testcase"):
            test_name = test.attrib["name"]
            skipped = any(child.tag == "skipped" for child in test)
            if test_name in always_skipped:
                always_skipped &= skipped
            else:
                always_skipped = skipped
```

**Comment by ogrisel** [comment]
Merged! Thanks very much for the fix @jeremiedbb.

**Pull request merged**